src/Keboola/StorageApi/Client.php

```diff
@@ -1253,18 +1256,52 @@ public function uploadFile($filePath, FileUploadOptions $options, FileUploadTran
             ->setSizeBytes(filesize($filePath))
             ->setFederationToken(true);
 
-        // 1. prepare resource
-        $result = $this->prepareFileUpload($newOptions);
+        $prepareResult = $this->prepareFileUpload($newOptions);
 
-        // 2. upload directly do S3 using returned credentials
-        // using federation token
-        $uploadParams = $result['uploadParams'];
+        switch ($prepareResult['provider']) {
+            case 'azure':
+                $this->uploadFileToAbs(
+                    $prepareResult,
+                    $filePath
+                );
+                break;
+            default:
+                $this->logger->warning('File upload without provider has been used', $prepareResult);
```

Comment:
Tohle bych asi klidně dal pryč, to je věc backedu že to nevrátil,

Reply:
To tam bylo jenom, kdyby to vracelo nějaký divný věci ze začátku, ale dám to pryč.